controller.py

```diff
@@ -446,64 +445,47 @@ def remove_items(self, collection_details):
         return feed_response(removal_feed)
 
     def register(self, do_get=HTTP.get_with_timeout):
-        opds_url = request.form.get('url')
-        if not opds_url:
-            return NO_OPDS_URL
-
-        AUTH_DOCUMENT_REL = AuthenticationForOPDSDocument.LINK_RELATION
-        auth_response = None
-
-        def get_auth_document(opds_feed):
-            links = opds_feed.get('feed', {}).get('links', [])
-            auth_links = [l for l in links if l.href and l.rel==AUTH_DOCUMENT_REL]
-            if not auth_links:
-                return None
-
-            auth_link = auth_links[0].get('href')
-            response = do_get(auth_link, allowed_response_codes=['2xx', '3xx'])
-            try:
-                return response.json()
-            except ValueError as e:
-                return None
+        public_key_url = request.form.get('url')
+        if not public_key_url:
+            return NO_AUTH_URL
 
         try:
             response = do_get(
-                opds_url, allowed_response_codes=['2xx', '3xx', 401]
+                public_key_url, allowed_response_codes=['2xx', '3xx']
             )
-            if response.status_code == 401:
-                # The feed requires authentication. This response should have
-                # the authentication document.
-                auth_response = response.json()
-            else:
-                feed = feedparser.parse(response.content)
-                auth_response = get_auth_document(feed)
         except Exception as e:
-            return INVALID_OPDS_FEED
+            return REMOTE_INTEGRATION_ERROR
+
+        content_type = None
+        if response.headers:
+            content_type = response.headers.get('Content-Type')
 
-        if not auth_response:
-            return AUTH_DOCUMENT_NOT_FOUND
+        if not (response.content and content_type == OPDS_2_MEDIA_TYPE):
+            # There's no JSON to speak of.
+            return INVALID_INTEGRATION_DOCUMENT
 
-        url = auth_response.get('id')
+        public_key_response = response.json()
+
+        url = public_key_response.get('id')
         if not url:
-            return INVALID_AUTH_DOCUMENT.detailed(
+            return INVALID_INTEGRATION_DOCUMENT.detailed(
                 "The OPDS authentication document is missing an id."
```

Comment:
You're not expecting to get an authentication document anymore. You should pick a name for the document you are expecting and use that in these messages.